### PR TITLE
Feature/clion licenses

### PR DESCRIPTION
**Pull request by strahlc**
dev-util/clion: add bindist restriction
dev-util/clion: add licenses for bundled libraries

Package-Manager: Portage-2.3.69, Repoman-2.3.16
Signed-off-by: Christian Strahl <c.a.strahl@gmail.com>

Closes: https://bugs.gentoo.org/694266

### CODE REVIEW

**Comment by mgorny** [comment]
How did you manage to find all those licenses? Did you just look at all files in licenses dir, or something more?

File: licenses/CDDL-1.1 [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+     DEATH OR PERSONAL INJURY RESULTING FROM SUCH PARTY'S NEGLIGENCE TO THE
+     EXTENT APPLICABLE LAW PROHIBITS SUCH LIMITATION. SOME JURISDICTIONS DO NOT
+     ALLOW THE EXCLUSION OR LIMITATION OF INCIDENTAL OR CONSEQUENTIAL DAMAGES,
+     SO THIS EXCLUSION AND LIMITATION MAY NOT APPLY TO YOU.
+
+8. U.S. GOVERNMENT END USERS.
+
+     The Covered Software is a "commercial item," as that term is defined in
+     48 C.F.R. 2.101 (Oct. 1995), consisting of "commercial computer software"
+     (as that term is defined at 48 C.F.R. ? 252.227-7014(a)(1)) and
```
Comment (ulm) [inline licenses/CDDL-1.1]: Broken encoding? This should be a section sign "§" not a question mark.
Where did you take this license file from?

Comment (strahlc) [reply licenses/CDDL-1.1]: I've rebased this feature branch to the current master, and dropped its CDDL-1.1 license file in favor of the ones in master (from commit https://gitweb.gentoo.org/repo/gentoo.git/commit/?id=9a8505e9128e8e80dc11b330e1b795ca465b9061)

File: dev-util/clion/clion-2019.2.1.ebuild [hunk 2]
```diff
@@ -9,11 +9,14 @@ DESCRIPTION="A complete toolset for C and C++ development"
 HOMEPAGE="https://www.jetbrains.com/clion"
 SRC_URI="https://download.jetbrains.com/cpp/CLion-${PV}.tar.gz -> ${P}.tar.gz"
 
-LICENSE="IDEA
-	|| ( IDEA_Academic IDEA_Classroom IDEA_OpenSource IDEA_Personal )"
+LICENSE="Apache-1.1 Apache-2.0 BSD BSD-2 CC0-1.0 CDDL-1.1 CPL-0.5 CPL-1.0
+	EPL-1.0 EPL-2.0 GPL-2 GPL-2-with-classpath-exception GPL-3 ISC
+	JDOM LGPL-2.1+ LGPL-3 MIT MPL-1.0 MPL-1.1 OFL oromatcher
+	public-domain PSF-2 sun-bcla-javahelp UoI-NCSA ZLIB
```
Comment (ulm) [inline dev-util/clion/clion-2019.2.1.ebuild]: What files in the package are licensed under sun-bcla-javahelp?
Javahelp used to be licensed under GPL-2-with-classpath-exception, and was recently relicensed to CDDL-1.1.

File: dev-util/clion/clion-2019.2.1.ebuild [hunk 3]
```diff
@@ -9,11 +9,14 @@ DESCRIPTION="A complete toolset for C and C++ development"
...
+	|| ( IDEA IDEA_Academic IDEA_Classroom IDEA_OpenSource IDEA_Personal )"
```
[the remark this answers is not in the document]
Comment (strahlc) [reply dev-util/clion/clion-2019.2.1.ebuild]: you're right, I've reordered the licenses

File: dev-util/clion/clion-2019.2.1.ebuild (hunk 2 above)
Comment (strahlc) [reply dev-util/clion/clion-2019.2.1.ebuild]: you're right, my fault.

**Comment by strahlc** [comment]
They've also removed the `oromatcher.jar`, but they still deliver `/opt/clion-2019.2.1/license/oromatcher_license.txt`.
Due to that I've removed oromatcher from the `LICENSE` list and removed the license file as well.

File: dev-util/clion/clion-2019.2.1.ebuild (hunk 2 above)
Comment (ulm) [reply dev-util/clion/clion-2019.2.1.ebuild]: oromatcher is gone as well? Was that for lib/oro-2.0.8.jar (which really is Apache-1.1)?

**Pull request closed without merging**